### PR TITLE
`azurerm_security_center_assessment_policy` : refactoring to use hashicorp/go-azure-sdk

### PR DESCRIPTION
**Pull request by sinbai**
Test Result:
```

=== RUN   TestAccServerVulnerabilityAssessment
--- PASS: TestAccServerVulnerabilityAssessment (267.42s)
=== RUN   TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy
    --- PASS: TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy (267.42s)
=== RUN   TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/basic
        --- PASS: TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/basic (80.27s)
=== RUN   TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/complete
        --- PASS: TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/complete (74.49s)
=== RUN   TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/update
        --- PASS: TestAccServerVulnerabilityAssessment/securityCenterAssessmentPolicy/update (112.67s)
PASS
```

### CODE REVIEW

File: internal/services/securitycenter/security_center_assessment_policy_resource.go [hunk 1]
```diff
@@ -142,57 +144,57 @@ func resourceArmSecurityCenterAssessmentPolicyCreate(d *pluginsdk.ResourceData,
...
-		params.AssessmentMetadataProperties.Categories = &categories
+		params.Properties.Categories = &categories
 	}
 
 	if v, ok := d.GetOk("threats"); ok {
-		threats := make([]security.Threats, 0)
+		threats := make([]assessmentsmetadata.Threats, 0)
 		for _, item := range v.(*pluginsdk.Set).List() {
-			threats = append(threats, (security.Threats)(item.(string)))
+			threats = append(threats, (assessmentsmetadata.Threats)(item.(string)))
```
Comment (tombuildsstuff) [inline internal/services/securitycenter/security_center_assessment_policy_resource.go:180]:
```suggestion
			threats = append(threats, assessmentsmetadata.Threats(item.(string)))
```

File: internal/services/securitycenter/security_center_assessment_policy_resource.go [hunk 2]
```diff
@@ -142,57 +144,57 @@ func resourceArmSecurityCenterAssessmentPolicyCreate(d *pluginsdk.ResourceData,
...
+			Severity:       assessmentsmetadata.Severity(d.Get("severity").(string)),
 		},
 	}
 
 	if v, ok := d.GetOk("categories"); ok {
-		categories := make([]security.Categories, 0)
+		categories := make([]assessmentsmetadata.Categories, 0)
 		for _, item := range v.(*pluginsdk.Set).List() {
-			categories = append(categories, (security.Categories)(item.(string)))
+			categories = append(categories, (assessmentsmetadata.Categories)(item.(string)))
```
Comment (tombuildsstuff) [inline internal/services/securitycenter/security_center_assessment_policy_resource.go:172]:
```suggestion
			categories = append(categories, assessmentsmetadata.Categories(item.(string)))
```

**Review by tombuildsstuff — APPROVED** [review]
A few minor things but otherwise 👍

File: internal/services/securitycenter/client/client.go [hunk 3]
```diff
@@ -5,13 +5,14 @@ package client
 
 import (
 	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security" // nolint: staticcheck
+	assessmentsmetadata_v2021_06_01 "github.com/hashicorp/go-azure-sdk/resource-manager/security/2021-06-01/assessmentsmetadata"
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/securitycenter/client/client.go]: Fixed.

File: internal/services/securitycenter/security_center_assessment_policy_resource_test.go [hunk 4]
```diff
@@ -72,21 +74,21 @@ func testAccSecurityCenterAssessmentPolicy_update(t *testing.T) {
...
+		if response.WasNotFound(resp.HttpResponse) {
 			return utils.Bool(false), nil
 		}
 
-		return nil, fmt.Errorf("retrieving Azure Security Center Assessment Metadata %q: %+v", state.ID, err)
+		return nil, fmt.Errorf("retrieving %q: %+v", *id, err)
 	}
 
-	return utils.Bool(resp.AssessmentMetadataProperties != nil), nil
+	return pointer.To(resp.Model != nil && resp.Model.Properties != nil), nil
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/securitycenter/security_center_assessment_policy_resource_test.go]: Fixed.

File: internal/services/securitycenter/security_center_assessment_policy_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/securitycenter/security_center_assessment_policy_resource.go]: Fixed.

File: internal/services/securitycenter/security_center_assessment_policy_resource.go (hunk 2 above)
Comment (sinbai) [reply internal/services/securitycenter/security_center_assessment_policy_resource.go:172]: Fixed.

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @sinbai

**Comment by sinbai** [comment]
> A few minor things but otherwise 👍

Hi @tombuildsstuff , thanks for your feedback. The code has been updated. Could you please take another look?

**Pull request merged**